geode-core/src/test/java/org/apache/geode/internal/process/FileProcessControllerIntegrationTest.java

```diff
@@ -0,0 +1,243 @@
+ */
+@Category(IntegrationTest.class)
+public class FileProcessControllerIntegrationTest {
+
+  private static final String STATUS_JSON = generateStatusJson();
+
+  private final AtomicReference<String> statusRef = new AtomicReference<>();
+
+  private File pidFile;
+  private File statusFile;
+  private File statusRequestFile;
+  private File stopRequestFile;
+  private int pid;
+  private FileControllerParameters params;
+  private ExecutorService executor;
+
+  @Rule
+  public ErrorCollector errorCollector = new ErrorCollector();
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Rule
+  public TestName testName = new TestName();
+
+  @Before
+  public void setUp() throws Exception {
+    ProcessType processType = ProcessType.LOCATOR;
+    File directory = temporaryFolder.getRoot();
+    pidFile = new File(directory, processType.getPidFileName());
+    statusFile = new File(directory, processType.getStatusFileName());
+    statusRequestFile = new File(directory, processType.getStatusRequestFileName());
+    stopRequestFile = new File(directory, processType.getStopRequestFileName());
+    pid = ProcessUtils.identifyPid();
+
+    params = mock(FileControllerParameters.class);
+    when(params.getPidFile()).thenReturn(pidFile);
+    when(params.getProcessId()).thenReturn(pid);
+    when(params.getProcessType()).thenReturn(processType);
+    when(params.getDirectory()).thenReturn(temporaryFolder.getRoot());
+
+    executor = Executors.newSingleThreadExecutor();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    assertThat(executor.shutdownNow()).isEmpty();
+  }
+
+  @Test
+  public void statusShouldAwaitTimeoutWhileFileIsEmpty() throws Exception {
+    // given: FileProcessController with empty pidFile
+    FileProcessController controller = new FileProcessController(params, pid, 10, MILLISECONDS);
+
+    // when:
+    verifyException(controller).status();
+
+    // then: we expect TimeoutException to be thrown
+    assertThat((Exception) caughtException()).isInstanceOf(TimeoutException.class)
+        .hasMessageContaining("Timed out waiting for process to create").hasNoCause();
+  }
+
+  @Test
+  public void statusShouldReturnJsonFromStatusFile() throws Exception {
+    // given: FileProcessController with pidFile containing real pid
+    new IntegerFileWriter(pidFile).writeToFile(pid);
+    FileProcessController controller = new FileProcessController(params, pid, 2, MINUTES);
+
+    // when: status is called in one thread
+    executor.execute(() -> {
+      try {
+        statusRef.set(controller.status());
+      } catch (Exception e) {
+        errorCollector.addError(e);
+      }
+    });
+
+    // and: json is written to the status file
+    new StringFileWriter(statusFile).writeToFile(STATUS_JSON);
+
+    // then: returned status should be the json in the file
+    await().until(() -> assertThat(statusRef.get()).isEqualTo(STATUS_JSON));
+  }
+
+  @Ignore // GEODE-3278
```

Comment:
This test has a few interesting things including the @Ignore for GEODE-3278 that warrant more documentation.